packages/blade/src/components/AutoComplete/_decisions/decisions.md

```diff
@@ -0,0 +1,240 @@
+
+```jsx
+<Dropdown selectionType="multiple">
+  <AutoComplete label="Select Cities" />
+  <DropdownOverlay>
+    <ActionList>
+      <ActionListItem title="Mumbai" value="mumbai" />
+      <ActionListItem title="Pune" value="pune" />
+      <ActionListItem title="Hyderabad" value="hyderabad" />
+    </ActionList>
+  </DropdownOverlay>
+</Dropdown>
+```
+
+### Controlled AutoComplete
+
+There are 3 controlled states that are relevant to AutoComplete
+
+1. Controlled state of filtering items
+2. Controlled state of the Input where user types
+3. Controlled state of selected items
+
+```jsx
+<AutoComplete
+  // For controlling the input value
+  value={}
+  onChange={(inputValue) => {}}
+  // For controlling the selections
+  selected={}
+  onSelectChange={({ name, values }) => {}}
+/>
+
+// ...
+
+// Filtering can be controlled by looping over <ActionListItem />
+<ActionList>
+  {cities.map((city) => <ActionListItem title={city} />)}
+</ActionList>
+```
+
+Complex flows such as [Adding New Item to ActionList](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=47091%3A711978&mode=dev) should be handled with Controlled API.
+
+<details>
+<summary>Show Full Code</summary>
+
+```jsx
+const allCities = ['Mumbai', 'Pune', 'Hyderabad'];
+
+const ControlledAutoComplete = () => {
+  // Controlled Filtering
+  const [cities, setCities] = React.useState(allCities);
+  // Controlled Input Value
+  const [currentInputValue, setCurrentInputValue] = React.useState('');
+  // Controlled Selections
+  const [currentSelections, setCurrentSelections] = React.useState<string[]>([]);
+
+  return (
+    <Dropdown>
+      <AutoComplete
+        label="Select City"
+        // Controlled input value
+        // Sets the input value of input element in AutoComplete
+        value={currentInputValue}
+        // Triggers when input value changes
+        onChange={(inputValue) => {
+          if (inputValue) {
+            // filtering logic
+            setCities(allCities.filter((city) => city.includes(inputValue)));
+          } else {
+            setCities(allCities);
+          }
+          setCurrentInputValue(inputValue);
+        }}
+        // Selected Items in AutoComplete
+        selected={currentSelections}
+        onSelectChange={({ values }) => {
+          setCurrentSelections(values);
+        }}
+      />
+      <DropdownOverlay>
+        <ActionList>
+          {cities.map((city) => (
+            <ActionListItem
+              title={city}
+              value={city.toLowerCase()}
+            />
+          ))}
+        </ActionList>
+      </DropdownOverlay>
+    </Dropdown>
+  );
+};
+```
+
+</details>
+
+### In BottomSheet
+
+> **Warning**
+>
+> We will have to turn BottomSheetHeader (and maybe BaseHeader) to a slot to allow adding Input element there.
+
+**Open Question:** Should we also remove `leading` `trailing` etc props from Header? since same can be implemented inside the slot. This is what we follow for Footer.
```

Comment:
Probably shouldn't do this, Header is already standardised across Modal & BottomSheet and any deviations will also needed to be carried to the BaseHeader. 

And anyways if people want to opt out of Header altogether they can just put things in the Body.

Reply:
So what will be the way to put input in header in this case? opt out of Header? or you mean we can go with API I proposed above where we keep leading trailing but also add children?

Reply:
The API you proposed is fine only. No other learning curve i can plug this right directly? and also do we need this? for eg: the ideal solution on mobile is to have an input and then open a bottomsheet right? so is this a valid pattern in dropdown's context? probably we can have 2 separate example to demo this? one with normal dropdown outside dropdown and one with dropdown inside?

Reply:
we now know right that in slot we'll only allow allow autocomplete so nothing else shall be removed as that can be added in slot that means it still remains as it is?

Reply:
Yupp we're going with what is proposed in doc. I'll remove this open question note.